server/ladder_service.py

```diff
@@ -134,6 +135,8 @@ async def update_data(self) -> None:
                 self.queues[queue_name].shutdown()
                 del self.queues[queue_name]
 
+        metrics.ladder_service_searches_dict_length.set(len(self._searches))
```

Comment:
I would suggest moving this to `LadderService.start_search` and `LadderService.cancel_search` for real-time updates.